Db client drivers (roque)

roquemolina opened the pull request:
Added Models, Routes & Bulks

roquemolina committed:
Models, some routes & User/Enterprise relation

The models were made according to what was analyzed in Miro (Please, review to see some changes). No validations have been included so far.
On the other hand, you would have to see some types of data. For example, "rating" is an array of numbers to later be able to get the percentage in double.
Added some routes to outline the CRUD of enterprise and users.
There are still tasks to be carried out, but it is a good idea to do a review of what has been coded so that the review is not too heavy.

roquemolina committed:
Added Models, Routes & Bulks

Routes: Enterprise, User, Driver, Reserve
Bulks: Enterprise, User, Driver
Added all models bc database.js imports all models to create relations

roquemolina committed:
Some fixs and reserve/trip relation

I made some small corrections and created the trip/reservation relationship.
Analyze the PR up to what has been done so far to confirm that it works.

manduinca commented on src/controllers/Enterprise/postEnterpriseController.js:
```diff
@@ -1,16 +1,15 @@
+    address,
+  });
+  return {
+    created: true,
```
I don't think it's necessary. We can say that the object was successfully created with http code 201.

roquemolina replied:
Ok. I'll resolve the conversation once the task is finished.

manduinca commented on src/controllers/Enterprise/putEnterpriseController.js:
```diff
@@ -1,18 +1,18 @@
-        ruc,
-    };
+  return {
+    updated: true,
```
Same here.

roquemolina replied:
Ok. I'll resolve the conversation once the task is finished.

manduinca commented on src/database.js:
```diff
@@ -6,56 +6,66 @@ const { DB_USER, DB_PASSWORD, DB_HOST, DB_PORT, DB_NAME } = process.env
+const StopModel = require('./models/Stop.js');
+const UserModel = require('./models/User.js');
+const FinanceDriverModel = require('./models/FinanceDriver.js');
+const FinanceEnterpriceModel = require('./models/FinanceEnterprice.js');
```
Enterprise

roquemolina replied:
Could you elaborate more? I don't understand what you mean.

manduinca commented on src/database.js:
```diff
@@ -6,56 +6,66 @@ const { DB_USER, DB_PASSWORD, DB_HOST, DB_PORT, DB_NAME } = process.env
 const sequelize = new Sequelize(`postgres://${DB_USER}:${DB_PASSWORD}@${DB_HOST}:${DB_PORT}/${DB_NAME}`, {
   logging: false,
   native: false
```
Why do we need it?

roquemolina replied:
- logging: A function that gets executed every time Sequelize would log something. Function may receive multiple parameters but only first one is printed by console.log. To print all values use (...msg) => console.log(msg). Set to console.log to see the raw SQL queries.
- native: A flag that defines if native library shall be used or not. Currently only has an effect for postgres. Lets Sequelize know we can use pg-native for ~ 30% more speed. In the case of 'pg' -- set this to true will allow SSL support.

manduinca commented on src/models/Detour.js:
```diff
@@ -0,0 +1,21 @@
+      allowNull: false
+    },
+  },
+  { timestamps: false });
```
Why is it necessary?

manduinca replied:
Well, I think we need to have timestamps enabled.

roquemolina replied:
- I set the allowNull to false so that the detour has to be described. Should it be disabled?
- Disable timeStamp to avoid filling with unnecessary data for the moment. If necessary we enable it.

manduinca commented on src/models/Detour.js:
Please, in models in general, we need to have the following fields:

- created_date
- created_by
- modified_date
- modified_by

These fields are important for auditory purposes.
One idea is to use a model base with these fields and extend it.

manduinca replied:
I saw that timestamp config does something similar to our requirement. I think it is sufficient.

roquemolina replied:
- The createdAt field will contain the timestamp representing the moment of creation, and the updatedAt will contain the timestamp of the latest update.
- created_by and modified_by should be added to the model as new input data. In which models do you think they should be added? All of them?

manduinca commented on src/models/Driver.js:
```diff
@@ -0,0 +1,73 @@
+    },
+    ruc: {
+      type: DataTypes.STRING,
+      allowNull: false
```
`allowNull` should be `True`.

roquemolina replied:
ok!

manduinca commented on src/models/Driver.js:
Please, we need to rethink this. I will leave some ideas on Miro.

manduinca commented on src/models/Reserve.js:
Comments are deprecated, that's the problem with unnecessary comments. It's better not to use comments, only when absolutely necessary.

manduinca commented on src/models/Stop.js:
```diff
@@ -0,0 +1,20 @@
+      type: DataTypes.STRING,
+      allowNull: false,
+    },
+    duration: {
```
Better `time` instead of `duration`. Also and more important, why not `TIME` DataType here?, It's a tricky problem, in Postgres it should be an `INTERVAL` DataType, but is possible to configure it from sequelize?
In v7 there is a custom data type so we can configure to `INTERVAL`, but I'm not sure that works in v6.

roquemolina replied:
There is no specific data type to represent a time interval directly as you would with a database management system that supports interval data types, such as PostgreSQL. Use two data types "hours" and "minutes" with integers to get you out of trouble.

roquemolina committed:
Models and implementation from enterprise to trip.

Sketch of the models and the implementation of the routes and services to create users, enterprises, drivers, cars, silvercars, reserves and trips.

Commit:
Add idea files to gitignore

manduinca committed:
Car domain logic done.

roquemolina committed:
Driver domain logic finished.
Driver account details in a separete model. (DriverAccount).

roquemolina committed:
Enterprise, Reserve, User and Trip domains added.

Trips and Reserve w/ notion parameters
and rating send info to passenget and driver

roquemolina committed:
All services and controllers updated with Car structure.

manduinca reviewed (approved):
LGTM!
